src/service.rs

```diff
@@ -524,11 +522,12 @@ impl Service {
     /// # Ok(())
     /// # }
     /// ```
-    pub fn start<S: AsRef<OsStr>>(&self, service_arguments: &[S]) -> Result<()> {
+    pub fn start<S: AsRef<OsStr>>(&self, service_arguments: &[S]) -> Result<(), ErrorKind> {
         let wide_service_arguments = service_arguments
             .iter()
-            .map(|s| WideCString::from_str(s).chain_err(|| ErrorKind::InvalidStartArgument))
-            .collect::<Result<Vec<WideCString>>>()?;
+            .map(|s| WideCString::from_str(s).map_err(|_| ErrorKind::InvalidStartArgument))
```

Comment:
If you use causes as I argued in the other thread, then you don't need to discard the error here. Instead it can be included inside `InvalidStartArgument` so the user can get a nice chain of errors. Explaining more in detail what went wrong and why.

Reply:
definitely, i'm hacking on this in phases. thanks for the feedback.